pkg/3scale/amp/operator/redis_reconciler.go

```diff
@@ -124,3 +133,29 @@ func Redis(apimanager *appsv1alpha1.APIManager, client client.Client) (*componen
 	}
 	return component.NewRedis(opts), nil
 }
+
+func (r *RedisReconciler) deleteSystemRedisSecretNamespaceKey() error {
+	secret := &corev1.Secret{}
+	err := r.Client().Get(context.TODO(), client.ObjectKey{Namespace: r.apiManager.Namespace, Name: "system-redis"}, secret)
+	if k8serr.IsNotFound(err) {
```

Comment:
Not a must, just another way of going through error handling in case when we want to handle isNotFound.
```
if err != nil && !k8serr.IsNotFound(err) {
       return err 
} else { 
      return nil 
}
```

Reply:
code after this will be unaccessable, as return nil. But yes, it could be written different way, sure. Thank you

Reply:
Good point, long day. 
```
if err != nil {
	if !k8serr.IsNotFound(err) {
		return err
	} else {
		return nil
	}
}
```